packages/salesforcedx-sobjects-faux-generator/src/generator/declarationGenerator.ts

```diff
@@ -0,0 +1,154 @@
+/*
+ * Copyright (c) 2021, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import { ChildRelationship, Field, SObject } from '../types';
+
+export const MODIFIER = 'global';
+export interface FieldDeclaration {
```

Comment:
@jag-sfdc The static representation could contain a list of these objects with some additional information. That's sufficient to generate apex and type files.